src/bucket/BucketManagerImpl.cpp

```diff
@@ -474,7 +485,19 @@ BucketManagerImpl::getBucketByHash(uint256 const& hash)
                    "BucketManager::getBucketByHash({}) found no bucket, making "
                    "new one",
                    binToHex(hash));
-        auto p = std::make_shared<Bucket>(canonicalName, hash);
+
+        std::optional<size_t> pageSize{};
+        size_t pageCutoff = 0;
+        if (mApp.getConfig().EXPERIMENTAL_BUCKET_KV_STORE)
+        {
+            pageSize = std::make_optional(
+                mApp.getConfig().EXPERIMENTAL_BUCKET_KV_STORE_INDEX_PAGE_SIZE);
+            pageCutoff =
+                mApp.getConfig().EXPERIMENTAL_BUCKET_KV_STORE_INDEX_CUTOFF;
+        }
+
+        auto p =
+            std::make_shared<Bucket>(canonicalName, hash, pageSize, pageCutoff);
```

Comment:
Looking at this more, it might be a good idea to have two bucket types, `Bucket` and `IndexedBucket`. `IndexedBucket` can just extend the base type with indexing and search related functions. It looks like all the paths that require the "definitely fast bucket" never do any searches with it and can be the base `Bucket` type. This makes the semantics a little more clear too, so we can have a `getBucketByHash` and a `getIndexedBucketByHash`.